### PR TITLE
Feat/notifications

### PR DESCRIPTION
**Pull request by karthik01455**
- Add functionality to create notifications in utils
- Add functionality to get notification by [projectId,memberId] (get notification for a person in project)
- Add functionality to edit notification 
- Add functionality to create notification when team Request is created.
- EmailId of tagged User is extracted from content
- Added pagination.
- Notification will be created automatically when celebration, team Request and announcement is created.
- Added query params to filter by readStatus and targetType(celebration, team Request and announcement).
- Test for controllers and services with 100% coverage with swagger documentation.

Screen shots:
GET
image
image
PUT
image

### CODE REVIEW

File: prisma/dashboard-db/schema.prisma [hunk 1]
```diff
@@ -126,6 +126,23 @@ model teamInformation {
...
+  content String?
+  projectId Int
+  createdAt DateTime @default(now())
+  targetType NotificationType?
+  targetId Int?
+  userNotifications userNotification[]
+}
+model userNotification {
+  id Int @id @unique @default(autoincrement())
+  memberId Int
```
Comment (sugumar-mani) [inline prisma/dashboard-db/schema.prisma:141]: member id and notification id should be unique

File: prisma/dashboard-db/schema.prisma [hunk 2]
```diff
@@ -126,6 +126,23 @@ model teamInformation {
...
+  readStatus Boolean @default(false)
+  notificationId Int
```
Comment (sugumar-mani) [inline prisma/dashboard-db/schema.prisma:143]: also notificationid should be a foreign key

File: prisma/dashboard-db/schema.prisma [hunk 3]
```diff
@@ -170,3 +187,9 @@ enum CelebrationType {
   IMPEDIMENT
 }
 
+enum NotificationType {
+    TEAMREQUEST
```
Comment (sugumar-mani) [inline prisma/dashboard-db/schema.prisma:191]: should be TEAM_REQUEST

Comment (karthik01455) [reply prisma/dashboard-db/schema.prisma]: fixed

File: prisma/dashboard-db/schema.prisma (hunk 1 above)
Comment (karthik01455) [reply prisma/dashboard-db/schema.prisma:141]: added

File: prisma/dashboard-db/schema.prisma (hunk 2 above)
Comment (karthik01455) [reply prisma/dashboard-db/schema.prisma:143]: relation is next line  144) notifications notification @relation(fields: [notificationId], references: [id], onDelete: Cascade)

**Pull request merged**